odoo/osv/expression.py

```diff
@@ -760,13 +760,13 @@ def child_of_domain(left, ids, left_model, parent=None, prefix=''):
                     for rec in left_model.browse(ids)
                 ])
                 if prefix:
-                    return [(left, 'in', left_model.search(doms).ids)]
+                    return [(left, 'in', left_model.search(doms, order='id').ids)]
```

Comment:
What I really wanted is to remove the 'ORDER BY' but the API doesn't allow it AFAIS. We could modify the method that generates the sql so it would ignore it when a False value is passed, but as we are in stable, sorting by id seems a good trade-off, it's still better to sort by id rather than the default order.

Reply:
We should rather do `order='id' if left_model._order else None` though.

Reply:
Makes sense, but this should then be extended to all `search()` calls executed during parsing, as they are only ever used for the purpose of generating the parameter for an 'in' or 'not in' clause, right?

Reply:
Probably, I admit I changed only this one because I had pathologically slow results with the specific use case of searching for `child_of` of locations and the profiling brought me here.

Now, I see that indeed we have similar search sub-queries for `many2one`, `many2many`, ... and it makes as much sense to do it there. I'll amend my commit.